File: app/src/main/java/did/delta/MainActivity.java
package did.delta;

import android.content.Intent;
import android.database.Cursor;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    private void refreshButton() {
        DBHelper helper = new DBHelper(MainActivity.this);
        Cursor c = helper.getReadableDatabase().rawQuery("select count(*) from " + DBHelper.SAVE_TABLE_NAME, null);
        if(!c.moveToFirst() || c.getInt(0) == 0) findViewById(R.id.buttonContinue).setEnabled(false);
        else findViewById(R.id.buttonContinue).setEnabled(true);
        c.close();
        helper.close();
    }

    @Override
    protected void onResume() {
        refreshButton();
        super.onResume();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.buttonContinue).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, GameActivity.class);
                intent.putExtra("load", true);
                startActivity(intent);
            }
        });
        refreshButton();
        findViewById(R.id.buttonNewGame).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this, GameActivity.class));
            }
        });
        findViewById(R.id.buttonExit).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });
        new Thread() {
            @Override
            public void run() {
                DBHelper helper = new DBHelper(MainActivity.this);
                helper.getWritableDatabase();
                helper.close();
            }
        }.start();
    }
}

File: app/src/main/java/did/delta/MainActivity.kt
package did.delta

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View

import did.delta.util.DBHelper

class MainActivity : AppCompatActivity() {

    private fun refreshButton() {
        // TODO: rewrite with many game types and buttons
        DBHelper(this@MainActivity).empty("save_normal")
    }

    override fun onResume() {
        refreshButton()
        super.onResume()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<View>(R.id.buttonContinue).setOnClickListener {
            val intent = Intent(this@MainActivity, GameActivity::class.java)
            intent.putExtra("save", true)
            startActivity(intent)
        }
        refreshButton()
        findViewById<View>(R.id.buttonNewGame).setOnClickListener {
            startActivity(Intent(this@MainActivity, GameActivity::class.java))
        }
        findViewById<View>(R.id.buttonWordsList).setOnClickListener {
            startActivity(Intent(this@MainActivity, ListActivity::class.java))
        }
        findViewById<View>(R.id.buttonExit).setOnClickListener {
            finish()
        }
        Thread {
            val helper = DBHelper(this@MainActivity)
            helper.getWritableDatabase()
            helper.close()
        }.start()
    }

}
